spark/src/main/java/org/opensearch/sql/spark/flint/FlintIndexMetadata.java

```diff
@@ -5,42 +5,22 @@
 
 package org.opensearch.sql.spark.flint;
 
-import java.util.Locale;
-import java.util.Map;
 import java.util.Optional;
+import lombok.Builder;
 import lombok.Data;
 
 @Data
+@Builder
 public class FlintIndexMetadata {
-  public static final String PROPERTIES_KEY = "properties";
-  public static final String ENV_KEY = "env";
-  public static final String OPTIONS_KEY = "options";
-
-  public static final String SERVERLESS_EMR_JOB_ID = "SERVERLESS_EMR_JOB_ID";
-  public static final String AUTO_REFRESH = "auto_refresh";
-  public static final String AUTO_REFRESH_DEFAULT = "false";
-
-  public static final String APP_ID = "SERVERLESS_EMR_VIRTUAL_CLUSTER_ID";
-  public static final String FLINT_INDEX_STATE_DOC_ID = "latestId";
 
   private final String jobId;
   private final boolean autoRefresh;
   private final String appId;
   private final String latestId;
+  private final FlintIndexStateModel indexStateModel;
```

Comment:
question? in case static method `getFlintIndexState(stateStore, datasourceName).apply(latestId.get());` can fetch IndexState, what is the major reason of code refactor?

Reply:
I was viewing it from cron job to clean all the indices of a datasource.
I was thinking to get IndexStateModel along with the IndexMetadata and this would be helpful in filtering out indices based on state.
Now I am implementing ALTER index and will probably remove this. Converting to draft for now to include alter index implementation as well.